backend/cmd/init.go

```diff
@@ -150,15 +152,37 @@ func InitCommand() *cobra.Command {
 					AdminUsername: uname,
 					AdminPassword: pword,
 				},
+				Timeout: timeout,
+			}
+
+			// Make sure at least one of the provided endpoints is reachable. This is
+			// required to debug TLS errors because the seeding below will not print
+			// the latest connection error (see
+			// https://github.com/sensu/sensu-go/issues/3663)
+			for _, url := range clientURLs {
+				tctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
+				defer cancel()
+				_, err = client.Status(tctx, url)
+				if err != nil {
+					// We do not need to log the error, etcd's client interceptor will log
+					// the actual underlying error
+					continue
+				}
+				// The endpoint did not return any error, therefore we can proceed
+				goto seed
 			}
+			// All endpoints returned an error, return the latest one
+			return err
 
+		seed:
 			return seedCluster(client, seedConfig)
 		},
 	}
 
 	cmd.Flags().String(flagInitAdminUsername, "", "cluster admin username")
 	cmd.Flags().String(flagInitAdminPassword, "", "cluster admin password")
 	cmd.Flags().Bool(flagInteractive, false, "interactive mode")
+	cmd.Flags().Uint(flagTimeout, defaultTimeout, "timeout, in seconds, for failing to establish a connection to etcd")
```

Comment:
I'd suggest making this a string, and then using viper's facility for parsing durations to get a `time.Duration`.

Viper just calls out to `time.ParseDuration` from stdlib, so the acceptable values are well-known.

It should avoid potential bugs around conversions. There is already technically a bug with using uint here, as converting a large enough uint, which is a 64-bit type on 64 bit platforms, to an int64 (time.Duration), will cause an overflow. Of course realistically, it's unlikely that anyone would try to specify such a large timeout. :laughing: